pkg/config/config_template.yaml

```diff
@@ -1,857 +1,1238 @@
+## Disable Python3 validation of python checks.
+#
 # disable_py3_validation: false
 
-# BETA: Encrypted Secrets
-#
-# This feature is in beta and its options or behaviour might break between
-# minor or bugfix releases of the Agent.
-#
-# The agent can call an external command to fetch secrets. The command will be
-# executed maximum once per instance containing an encrypted password.
-# Secrets are cached by the agent, this will avoid executing again the
-# secret_backend_command to fetch an already known secret (useful when combine
-# with Autodiscovery). This feature is still in beta.
-#
-# For more information see: https://github.com/DataDog/datadog-agent/blob/master/docs/agent/secrets.md
+## @param secret_backend_command - string - optional
+## `secret_backend_command` is the path to the script to execute to fetch secrets.
+## The executable must have specific rights that differ on Windows and Linux.
+##
+## For more information see: https://github.com/DataDog/datadog-agent/blob/master/docs/agent/secrets.md
 #
-# Path to the script to execute. The script must belong to the same user used
-# to run the agent. Executable right must be given to the agent and no rights
-# for 'group' or 'other'.
-# secret_backend_command: /path/to/command
+# secret_backend_command: <COMMAND_PATH>
+
+## @param secret_backend_arguments - list of strings - optional
+## If secret_backend_command is set, specify here a list of arguments to give to the command at each run.
 #
-# A list of arguments to give to the command at each run (optional)
 # secret_backend_arguments:
-#   - argument1
-#   - argument2
+#   - <ARGUMENT_1>
+#   - <ARGUMENT_2>
+
+## @param secret_backend_output_max_size - integer - optional - default: 1024
+## The size in bytes of the buffer used to store the command answer (apply to both stdout and stderr)
 #
-# The size in bytes of the buffer used to store the command answer (apply to
-# both stdout and stderr)
 # secret_backend_output_max_size: 1024
+
+## @param secret_backend_timeout - integer - optional - default: 5
+## The timeout to execute the command in second
 #
-# The timeout to execute the command in second
 # secret_backend_timeout: 5
-#
+
 {{ end -}}
 {{- if .LogsAgent }}
-# Logs agent
+
+###################################
+## Logs Collection Configuration ##
+###################################
+
+## @param logs_enabled - boolean - optional - default: false
+## Enable Datadog Agent logs collection by setting logs_enabled to true.
 #
-# Enable logs collection, disabled by default
 # logs_enabled: false
+
+## @param logs_config - custom object - optional
+## Enter specific configurations for your Log collection.
+## Uncomment this parameter and the one below to enable them.
+## See https://docs.datadoghq.com/logs/log_collection/
 #
 # logs_config:
+
+  ## @param container_collect_all - boolean - optional - default: false
+  ## Enable container log collection for all the containers (see ac_exclude to filter out containers)
```

Comment:
nits: sometimes you use `logs collection`, sometimes `log collection`, let's stick to one